RULES.md

```diff
@@ -1,536 +1,1037 @@
-| [`ThreadExecutionError`](#ThreadExecutionError)                         	| ExecutionException during multithreaded validation     	|
-| [`URISyntaxError`](#URISyntaxError)                                     	| A string could not be parsed as a URI reference.       	|
+| System error name                                                         | Description                                            	| Notice Code (as it appears in the `system_errors.json` file) |
+|-------------------------------------------------------------------------	|--------------------------------------------------------	|--------------------------------------------------------------|
+| [`IOError`](#IOError)                                                   	| Error in IO operation.                                 	| i_o_error                                                    |
+| [`RuntimeExceptionInLoaderError`](#RuntimeExceptionInLoaderError)       	| RuntimeException while loading GTFS dataset in memory. 	| runtime_exception_in_loader_error                            |
+| [`RuntimeExceptionInValidatorError`](#RuntimeExceptionInValidatorError) 	| RuntimeException while validating GTFS archive.        	| runtime_exception_in_validator_error                         |
+| [`ThreadExecutionError`](#ThreadExecutionError)                         	| ExecutionException during multithreaded validation     	| thread_execution_error                                       |
+| [`URISyntaxError`](#URISyntaxError)                                     	| A string could not be parsed as a URI reference.       	| u_r_i_syntax_error                                           |
 
-## Notices
+## Errors
 
-### Errors
+<a name="BlockTripsWithOverlappingStopTimesNotice"/>
 
-#### BlockTripsWithOverlappingStopTimesNotice
+### BlockTripsWithOverlappingStopTimesNotice
 
 Trips with the same block id have overlapping stop times.
-
-##### References:
+<details>
+
+#### Notice fields description
+| Field name      	| Description                               	            | Type   	|
+|-----------------	|---------------------------------------------------------	|--------	|
+| `csvRowNumberA` 	| The row number from `trips.txt` of the first faulty trip. | Long   	|
+| `tripIdA`       	| The id of first faulty trip.                           	| String 	|
+| `serviceIdA`    	| The service id of the first faulty trip.                 	| String 	|
+| `csvRowNumberB` 	| The row number from `trips.txt` of the second faulty trip.| Long   	|
+| `tripIdB`       	| The id of the other faulty trip.             	            | String 	|
+| `serviceIdB`    	| The service id of the other faulty trip.     	            | String 	|
+| `blockId`       	| The `trips.block_id` of the overlapping trip.	            | String 	|
+| `intersection`  	| The overlapping period.                      	            | Date   	|
+
+#### Affected files
```

Comment:
I'm torn as to whether we should include "Affected files", and maybe even "References", in the details section. I think a frequent use case is for the validator to flag an error, a producer or consumer to look up the rule, and then want to verify for themselves that the text in the GTFS spec actually requires/prohibits something - so they want a direct hyperlink to the section of the spec that covers this rule.

Maybe keep "Affected files" outside details, but leave notice JSON fields table and "References" inside details?

Reply:
Good point. 
I took out the _References_ of the _details_ section to make it more accessible, because for all file requirement notices (such as `empty_column_name`) or best practices (such as `missing_feed_info_date`), the `References` section is more relevant. 
The references to the python validator should be replaced with the reference to the corresponding portion of the spec, but this will have to be in another PR!